pages/studentPortal/web/React/tailwindcss-gridflex.tsx

```diff
@@ -0,0 +1,93 @@
+import { useRouter } from "next/router";
+import React from "react";
+import LessonComponent, {
+  LessonComponentData,
+} from "../../../../components/coding/studentPortal/LessonComponent";
+import { Button } from "../../../../components/ui/Button";
+
+const TailwindGridFlex = ({ lessonComponents }) => {
+  const router = useRouter();
+
+  const handleContinue = () => {
+    router.push("/studentPortal/web/React/tailwindcss-colourstyling");
+  };
+  return (
+    <>
+      <div className="grid grid-cols-1 gap-8 px-4 pt-4 m-8 sm:px-12">
+        {lessonComponents.map((it, index) => (
+          <LessonComponent data={it} key={index} />
+        ))}
+      </div>
+      <div className="flex my-8 mr-8 sm:justify-end">
+        <Button onClick={handleContinue} label="Continue" />
+      </div>
+    </>
+  );
+};
+
+export async function getServerSideProps({ params }) {
+  const lessonComponents: LessonComponentData[] = [
+    {
+      component: "title",
+      text: "Tailwind Lesson 1: Grid & Flexbox",
+    },
+    {
+      component: "resource-list",
+      resources: [
+        {
+          title: "Official TailwindCSS Documentation",
+          description:
+            "The official TailwindCSS documentation. Use CTRL+K to quick search for styling classes.",
+          image:
+            "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d5/Tailwind_CSS_Logo.svg/1200px-Tailwind_CSS_Logo.svg.png?20211001194333",
+          link: "https://tailwindcss.com",
+        },
+        {
+          title: "HTML TailwindCSS Easy Setup",
+          description:
+            "Configure TailwindCSS for an HTML document to practice using tailwind clsses. To do this, complete step one in the installation setup instructions under Play CDN.",
+          image:
+            "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d5/Tailwind_CSS_Logo.svg/1200px-Tailwind_CSS_Logo.svg.png?20211001194333",
+          link: "https://tailwindcss.com/docs/installation/play-cdn",
+        },
+      ],
+    },
+    {
+      component: "description",
+      text: "TailwindCSS is used to style webpages. Similar to inline CSS, Tailwind classes are written directly in the `className` attribute. The official tailwindcss documentation linked above is well designed and thorough, but can feel overwhelming for new users. We recommend that as you read this lesson, you solidify what you have learned by exploring the tailwind documentation.",
+    },
+    {
+      component: "code-sandbox",
+      title: "Grid Columns Example",
+      link: "https://codesandbox.io/embed/cocky-surf-2x7k1c?fontsize=14&hidenavigation=1&theme=dark",
+    },
+    {
+      component: "description",
+      text: "In general, grid is used to describe the outer level elements of the page. The code sandbox example above demonstrates how to use grid-cols-5 on line 5, to create a table with 5 columns. .",
```

Comment:
I don't think your explanation of grid vs. flex is accurate and / or clear enough.

Try reading through a couple articles on it... here's one: https://www.geeksforgeeks.org/comparison-between-css-grid-css-flexbox/

I'd consider adding one of these to the resource list as well